mrack.spec

```diff
@@ -1,59 +1,145 @@
 Requires:       python3-click
 Requires:       python3-pyyaml
-Requires:       testcloud
 Requires:       sshpass
 
+%package -n     python3-%{name}-aws
+Summary:        AWS provider plugin for mrack
+Requires:       python3-%{name}lib == %{version}-%{release}
+Requires:       python3-boto3
+Requires:       python3-botocore
+
+%package -n     python3-%{name}-beaker
+Summary:        Beaker provider plugin for mrack
+Requires:       python3-%{name}lib == %{version}-%{release}
+Requires:       beaker-client
+
+%package -n     python3-%{name}-openstack
+Summary:        Openstack provider plugin for mrack
+Requires:       python3-%{name}lib == %{version}-%{release}
+Recommends:     python3-AsyncOpenStackClient
+
+%package -n     python3-%{name}-podman
+Summary:        Podman provider plugin for mrack
+Requires:       python3-%{name}lib == %{version}-%{release}
+Requires:       podman
+
+%package -n     python3-%{name}-virt
+Summary:        Virtualization provider plugin for mrack using testcloud
+Requires:       python3-%{name}lib == %{version}-%{release}
+Requires:       testcloud
+
+%description        cli
+%{name}-cli contains mrack command which functionality can be extended by installing mrack plugins
+
+%description -n     python3-%{name}lib
+python3-%{name}lib contains core mrack functionalities and static provider which can be used as a library
+
+%description -n     python3-%{name}-aws
+%{name}-aws is an additional plugin with AWS provisioning library extending mrack package
+
+%description -n     python3-%{name}-beaker
+%{name}-beaker is an additional plugin with Beaker provisioning library extending mrack package
+
+%description -n     python3-%{name}-openstack
+%{name}-openstack is an additional plugin with OpenStack provisioning library extending mrack package
+
+%description -n     python3-%{name}-podman
+%{name}-podman is an additional plugin with Podman provisioning library extending mrack package
+
+%description -n     python3-%{name}-virt
+%{name}-virt is an additional plugin with Virualization provisioning library extending mrack package using testcloud
+
 %prep
-%autosetup -n %{srcname}-%{version}
+%autosetup -p1 -n %{name}-%{version}
 # Remove bundled egg-info
-rm -rf %{srcname}.egg-info
+rm -rf %{name}.egg-info
 
 %build
 %py3_build
 
 %install
 %py3_install
 
-%files -n %{srcname}
+%files
+%license LICENSE
+%doc README.md
+
+%files cli
+%license LICENSE
+%doc README.md
+%{_bindir}/%{name}
+
+%files -n python3-%{name}lib
 %license LICENSE
 %doc README.md
-%{_bindir}/mrack
-%{python3_sitelib}/%{srcname}
-%{python3_sitelib}/%{srcname}-%{version}-py%{python3_version}.egg-info
+%{python3_sitelib}/%{name}
+%{python3_sitelib}/%{name}-%{version}-py%{python3_version}.egg-info
+%exclude %{python3_sitelib}/%{name}/providers/utils/{,__pycache__/}osapi.py
+%exclude %{python3_sitelib}/%{name}/providers/{,__pycache__/}{%{provider_plugins}}.*
+%exclude %{python3_sitelib}/%{name}/transformers/{,__pycache__/}{%{provider_plugins}}.*
```

Comment:
I'd personally do `%dir %{python3_sitelib}/%{name}` and then drop the %exclude statements

Reply:
As discussed this would create a long list of includes doing it this way... Lets keep the excludes only ?

Reply:
taking back this one, the list of "include" statements would be too large...